services/static-webserver/client/source/class/osparc/data/model/Study.js

```diff
@@ -553,80 +578,82 @@ qx.Class.define("osparc.data.model.Study", {
       return !this.getUi().getSlideshow().isEmpty();
     },
 
-    serializeStudyData: function() {
-      let studyData = {};
-      const propertyKeys = this.self().getProperties();
-      propertyKeys.forEach(key => {
-        if (key === "workbench") {
-          studyData[key] = this.getWorkbench().serialize();
-          return;
-        }
-        if (key === "ui") {
-          studyData[key] = this.getUi().serialize();
-          return;
-        }
-        const value = this.get(key);
-        studyData[key] = value;
-      });
-      return studyData;
-    },
-
-    serialize: function(clean = true) {
-      let jsonObject = {};
-      const propertyKeys = this.self().getProperties();
-      propertyKeys.forEach(key => {
-        if (this.self().IgnoreSerializationProps.includes(key)) {
-          return;
-        }
-        if (key === "workbench") {
-          jsonObject[key] = this.getWorkbench().serialize(clean);
-          return;
-        }
-        if (key === "ui") {
-          jsonObject[key] = this.getUi().serialize();
-          return;
-        }
-        const value = this.get(key);
-        if (value !== null) {
-          // only put the value in the payload if there is a value
-          jsonObject[key] = value;
-        }
-      });
-      return jsonObject;
-    },
-
-    patchStudy: function(fieldKey, value) {
+    patchStudy: function(studyChanges) {
       return new Promise((resolve, reject) => {
-        const patchData = {};
-        patchData[fieldKey] = value;
         const params = {
           url: {
             "studyId": this.getUuid()
           },
-          data: patchData
+          data: studyChanges
         };
         osparc.data.Resources.fetch("studies", "patch", params)
           .then(() => {
-            const upKey = qx.lang.String.firstUp(fieldKey);
-            const setter = "set" + upKey;
-            this[setter](value);
+            Object.keys(studyChanges).forEach(fieldKey => {
+              const upKey = qx.lang.String.firstUp(fieldKey);
+              const setter = "set" + upKey;
+              this[setter](studyChanges[fieldKey]);
+            })
             // A bit hacky, but it's not sent back to the backend
             this.set({
               lastChangeDate: new Date()
             });
-            const studyData = this.serializeStudyData();
+            const studyData = this.serialize();
             resolve(studyData);
           })
           .catch(err => reject(err));
       });
     },
 
-    updateStudy: function(params, run = false) {
+    /**
+     * Call patch Study, but the changes were already applied on the frontend
+     * @param studyDiffs {Object} Diff Object coming from the JsonDiffPatch lib. Use only the keys, not the changes.
+     */
+    patchStudyDelayed: function(studyDiffs) {
+      return new Promise((resolve, reject) => {
```

Comment:
No need for a new promise that breaks the promise chain... just `return Promise.all(promises).then(..).catch(..)` and inside the `then`, just `return studyData` instead of `resolve(studyData)`

Reply:
Awesome!!